src/python/WMCore/WMBS/File.py

```diff
@@ -399,7 +399,7 @@ def setLocation(self, pnn, immediateSave=True):
         locations could be added - confusing when file requires locations on its
         first creation (breaks transaction model in Fileset commits etc)
         """
-        if isinstance(pnn, str):
+        if isinstance(pnn, (str, bytes)):
```

Comment:
Did you have to add this because the pnn can be bytes? would it make sense to convert it to a unicode string with our beloved `decodeBytesToUnicodeConditional(, PY3)`? Or do we want to keep it of bytes type?

Reply:
I discussed this with Dario over slack. I made it clear in the original description that this is needed if we are running WMAgent in a python2 stack.
Also, the main concern was to perform the correct set operation (regardless of a unicode or byte string).
Thanks for your input, Dario!

Reply:
Perfect! Thank you Alan! I am marking this as resolved